mne/preprocessing/_beer_lambert_law.py

```diff
@@ -0,0 +1,119 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Eric Larson <larson.eric.d@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+
+import re as re
+import numpy as np
+
+import mne
+from .. import rename_channels
+from ..io import BaseRaw
+from ..io.pick import _picks_to_idx
+from ..io.constants import FIFF
+from ..utils import _validate_type
+
+
+def beer_lambert_law(raw, ppf=0.1):
+    r"""Convert NIRS optical density data to haemoglobin concentration.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The optical density data.
+    ppf : float
+        The partial pathlength factor.
+
+    Returns
+    -------
+    raw : instance of Raw
+        The modified raw instance.
+    """
+    raw = raw.copy().load_data()
+    _validate_type(raw, BaseRaw, 'raw')
+
+    freqs = np.unique(_channel_frequencies(raw))
+    picks = _check_channels_ordered(raw, freqs)
+    abs_coef = _load_absorption(freqs)
+    distances = raw._probe_distances()
+
+    for ii in picks[::2]:
+
+        EL = abs_coef * distances[ii] * ppf
+        iEL = np.linalg.pinv(EL)
+
+        raw._data[[ii, ii + 1]] = (raw._data[[ii, ii + 1]].T * iEL.T).T * 1e3
+
+        raw.info['chs'][ii]['coil_type'] = FIFF.FIFFV_COIL_FNIRS_HBO
+        raw.info['chs'][ii + 1]['coil_type'] = FIFF.FIFFV_COIL_FNIRS_HBR
+
+        rename_channels(raw.info, {raw.info['chs'][ii]['ch_name']:
+                                   raw.info['chs'][ii]['ch_name'][:-4] +
+                                   ' hbo'})
+        rename_channels(raw.info, {raw.info['chs'][ii + 1]['ch_name']:
+                                   raw.info['chs'][ii + 1]['ch_name'][:-4] +
+                                   ' hbr'})
+
+    return raw
+
+
+def _channel_frequencies(raw):
+    """Return the light frequency for each channel."""
+    picks = _picks_to_idx(raw.info, 'fnirs_od')
+    freqs = np.empty_like(picks)
```

Comment:
So always int? `picks` will be `int`. Probably better to be explicit with dtype here like:
```suggestion
    freqs = np.empty(picks.size, int)
```